### PR TITLE
CLDR-14954 Add Abkhazian locale to CLDR

### PR DESCRIPTION
**Pull request by danielinux7**
CLDR-14954

 - [x] This PR completes the ticket.
 - [x] Updated PR title and link in previous line to include Issue number.

### CODE REVIEW

**Comment by srl295** [comment]
Followon from #1392

**Comment by danielinux7** [comment]
`Error:  (TestBasic.java:815) : likelyMax(locale=ab_JO) == likelyMax(defaultContent=ab): expected "ab_Cyrl_GE", got "ab_Cyrl_JO"`
`java.lang.InternalError: SupplementalData.defaultContentToChild(): defaultContent list in wrong order? Tried to map ab -> ab_JO, replacing ab_GE `
Hello,
@srl295 would you have an idea why I am getting these errors?

File: common/supplemental/attributeValueValidity.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 			<!-- This list contains ONLY those languages considered to be 'of interest' as part of CLDR, -->
 			<!-- and is maintained manually by the CLDR TC -->
 			<variable id='$language' type='choice'>
-				af agq ak am ar as asa ast az
+				ab af agq ak am ar as asa ast az
```
[the remark this answers is not in the document]
Comment (srl295) [reply common/supplemental/attributeValueValidity.xml:13]: @macchiati  might know. This is all probably too late for cldr v40

**Review by srl295 — DISMISSED** [review]
looking better

**Comment by danielinux7** [comment]
@srl295 I am getting these errors:
`Error:  (TestBasic.java:815) : likelyMax(locale=ab_TR) == likelyMax(defaultContent=ab): expected "ab_Cyrl_GE", got "ab_Cyrl_TR"`

`Error:  (TestInheritance.java:604) likely subtag inconsistent with default contents: 
	ab =>( dc) ab_TR
	ab => (ls) ab_Cyrl_GE`

**Comment by srl295** [comment]
> @srl295 I kept ab_TR but I am still getting these errors:
> `Error: (TestBasic.java:815) : likelyMax(locale=ab_TR) == likelyMax(defaultContent=ab): expected "ab_Cyrl_GE", got "ab_Cyrl_TR"`
> 
> `Error: (TestInheritance.java:604) likely subtag inconsistent with default contents: ab =>( dc) ab_TR ab => (ls) ab_Cyrl_GE`
> 
> I think it has to do with this file:
> cldr/common/supplemental/likelySubtags.xml

Probably due to population (?) it may consider `GE` a more likely defaultContent.  Is that possible? should ab_GE be the default content?

**Comment by danielinux7** [comment]
> Probably due to population (?) it may consider GE a more likely defaultContent. Is that possible? should ab_GE be the default content?

defaultContent should be ab_AB, Abkhazia which is partially a recognized territory, but not getting into this mess right now, I want to just add the language to v40.
Let defaultContent be just ab without territory.

**Comment by srl295** [comment]
> > Probably due to population (?) it may consider GE a more likely defaultContent. Is that possible? should ab_GE be the default content?
> 
> defaultContent should be ab_AB, Abkhazia which is partially a recognized territory, but not getting into this mess right now, I want to just add the language to v40.
> Let defaultContent be just ab without territory.

needs a territory.  can you set `ab_GE`?  

and as i said this might not make v40 but perhaps right after

**Comment by danielinux7** [comment]
> needs a territory. can you set ab_GE?

Done.

> and as i said this might not make v40 but perhaps right after

Okay, I'll do what I can from my end to update the survey tool with the required data. It would be nice to have it in v40.

The test is passing.

**Comment by danielinux7** [comment]
@srl295 
Hello Steven,
I made the changes and the tests are passing.

File: common/supplemental/attributeValueValidity.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (srl295) [reply common/supplemental/attributeValueValidity.xml:13]: @danielinux7 why was `ab` moved here?

**Comment by srl295** [comment]
@danielinux7 I'm rerunning tests

File: common/supplemental/supplementalData.xml [hunk 2]
```diff
@@ -1272,7 +1272,7 @@ XXX Code for transations where no currency is involved
 		<language type="aa" scripts="Latn"/>
 		<language type="aa" territories="DJ ET" alt="secondary"/>
 		<language type="ab" scripts="Cyrl"/>
-		<language type="ab" territories="GE" alt="secondary"/>
+		<language type="ab" territories="GE TR" alt="secondary"/>
```
Comment (pedberg-icu) [inline common/supplemental/supplementalData.xml:1275]: Hmm, this is data that is generated. To keep this chnage from getting lost we may need the fix discussed in https://unicode-org.atlassian.net/browse/CLDR-13595?focusedCommentId=162212 (and see the comments preceding that for context)

File: common/supplemental/attributeValueValidity.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielinux7) [reply common/supplemental/attributeValueValidity.xml:13]: @srl295 I looked at what others did to add a new locale and I did the same.
This is one of the files they modified.
https://unicode-org.atlassian.net/browse/CLDR-13901

File: common/supplemental/supplementalData.xml (hunk 2 above)
Comment (danielinux7) [reply common/supplemental/supplementalData.xml:1275]: @srl295 @pedberg-icu
I am afraid this is going to take more time.
How about we just keep ab_GE, that wouldn't require changes to this file, it's already there.
I'll add ab_TR later.

**Comment by danielinux7** [comment]
@srl295 @pedberg-icu 
I removed ab_TR.
Is this looking good?

**Review by srl295 — APPROVED** [review]

**Comment by srl295** [comment]
tests are passing. I've asked the CLDR-TC about approving the ticket.

**Comment by srl295** [comment]
@pedberg-icu look OK to you now?

**Comment by pedberg-icu** [comment]
Yes, looks good to me.

**Pull request merged**